src/Raven.Server/Documents/Handlers/Debugging/NodeDebugHandler.cs

```diff
@@ -103,79 +107,152 @@ await using (var writer = new AsyncBlittableJsonTextWriterForDebug(context, Serv
                 writer.WritePropertyName("Result");
 
                 writer.WriteStartArray();
+
                 while (tasks.Count > 0)
                 {
                     var task = await Task.WhenAny(tasks);
                     tasks.Remove(task);
                     context.Write(writer, task.Result.ToJson());
+
+                    if (setStatusCodeOnError && task.Result.HasErrors)
+                        HttpContext.Response.StatusCode = (int)HttpStatusCode.ServiceUnavailable;
+
                     if (tasks.Count > 0)
-                    {
                         writer.WriteComma();
-                    }
+
                     await writer.MaybeFlushAsync();
                 }
+
                 writer.WriteEndArray();
                 writer.WriteEndObject();
             }
         }
 
+        private class SetupAliveCommand : RavenCommand
+        {
+            public override HttpRequestMessage CreateRequest(JsonOperationContext ctx, ServerNode node, out string url)
+            {
+                url = $"{node.Url}/setup/alive";
+
+                var request = new HttpRequestMessage
+                {
+                    Method = HttpMethod.Get,
+                };
+
+                return request;
+            }
+        }
+
         internal class PingResult : IDynamicJsonValueConvertible
         {
             public string Url;
+            public SetupAliveInfo SetupAlive;
```

Comment:
In our *other system* we check for top-level Error property, which I see has gone missing. Can we keep it in some shape?

Reply:
Talked with @grisha-kotler. We'll work on adjusting the calling code to new response shape.